application/src/main/java/org/opentripplanner/service/streetdecorator/model/EdgeLevelInfo.java

```diff
@@ -0,0 +1,12 @@
+package org.opentripplanner.service.streetdecorator.model;
+
+public record EdgeLevelInfo(VertexLevelInfo lowerVertexInfo, VertexLevelInfo upperVertexInfo) {
+  public boolean matchesNodes(long firstOsmVertexId, long secondOsmVertexId) {
```

Comment:
Add javadoc.

Reply:
I changed the function to `canBeAppliedToEdge(Edge edge)` to make it more understandable. I also added documentation